Lib/fontTools/fontBuilder.py

```diff
@@ -486,15 +486,12 @@ def setupOS2(self, **values):
         """Create a new `OS/2` table and initialize it with default values,
         which can be overridden by keyword arguments.
         """
-        if "xAvgCharWidth" not in values:
-            gs = self.font.getGlyphSet()
-            widths = [
-                gs[glyphName].width
-                for glyphName in gs.keys()
-                if gs[glyphName].width > 0
-            ]
-            values["xAvgCharWidth"] = int(round(sum(widths) / float(len(widths))))
         self._initTableWithValues("OS/2", _OS2Defaults, values)
+        if "xAvgCharWidth" not in values:
+            assert (
+                "hmtx" in self.font
+            ), "the 'hmtx' table must be setup before the 'OS/2' table"
+            self.font["OS/2"].recalcXAvgCharWidth(self.font)
```

Comment:
This adds a dependency on `hmtx` table that this function didn’t have before, it might break some existing code but since the fix would probably only require calling `setupHorizontalMetrics()` first, I think it is OK.

Reply:
it already depeneded implicitly on hmtx being present, because it was getting a _TTGlyphSet which requires it

Reply:
I see, thanks.